### PR TITLE
Add Sphinx documentation auto-generation in Travis

### PR DESCRIPTION
**Pull request by creffett**
Will need some configuration on Travis to let it auto-push the gh-pages branch, see https://github.com/Syntaf/travis-sphinx#obtaining-a-personal-access-token.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,7 +14,12 @@ before_install:
     #- cat install.txt | xargs sudo apt-get install -y
   - sudo apt-get install -y clang-3.9 clang-format-3.9 clang-tidy-3.9 flawfinder libxml2 pylint python-git python-yapsy findbugs
   - sudo -H pip install -r requirements.txt
+  - sudo pip install sphinx travis-sphinx
```
Comment (denewiler) [inline .travis.yml]: These should be added to `requirements.txt`.

**Comment by denewiler** [comment]
I added a personal access token for gh-pages and linked it to the Travis job. We should be good on that side.

File: .travis.yml (hunk 1 above)
Comment (denewiler) [reply .travis.yml]: Actually, I can see these being separate since they are only used by Travis, not by typical developers.

**Comment by tdenewiler** [comment]
@creffett, looking good.

https://sscpac.github.io/statick/

**Pull request merged**